### PR TITLE
Create user interface

### PR DESCRIPTION
**Pull request by ekenecf**
Create an app.rb file that will serve as my console app entry-point. It should have methods that do the following:
List all books.
List all people.
Create a person (teacher or student, not a plain Person).
Create a book.
Create a rental.
List all rentals for a given person id.

In my main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:
Present the user with a list of options to perform.
Lets users choose an option.
If needed, ask for parameters for the option.
Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
+require_relative './book'
```
Comment (rustam-tolipov) [inline app.rb:1]: - [ ] You made great effort on this project! However your app is not running properly. It's automatically exiting when we run the app. Please fix this issue to make your project testable.


As you can see in this video:

https://user-images.githubusercontent.com/69011963/177010593-88ea1bea-833b-4e9b-8e60-ae15ecfd270f.mov

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+require_relative './rental'
+
+class App
+  def initialize()
+    @book_list = []
+    @rentals = []
+    @person = []
+  end
+
+  def menu
```
Comment (rustam-tolipov) [inline app.rb:14]: - [ ] [OPTIONAL] Kindly add MIT license to your project. When MIT license link clicked on readme. It's showing broken link.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @ekenecf, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    puts "'Has parent permission? [Y/N]: '"
+    permission = gets.chomp
+    if permission == 'N'
+      student = Student.new(age, name, permission: false)
+      @person.push(student)
+    else
+      student = Student.new(age, name, permission: true)
+    end
+    @person.push(student)
+  end
```
Comment (keneogu) [inline app.rb]: I'm not able to create student so looking at your person.rb file. you have parent_permission not permission. change `permission: true` to` parent_permission: true`

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+  def create_person
+    print "\n Do you want to create a student (1) or a teacher (2)? [Input the number]: "
+    user_input = gets.chomp
+    case user_input
+    when '1'
+      create_student
+    when '2'
+      create_teacher
```
Comment (keneogu) [inline app.rb:66]: There's no create_teacher method, I'm not able to create teacher. please add the `create_teacher` method.

**Review by keneogu — CHANGES_REQUESTED** [review]
Hi @ekenecf,

Good job so far!

- All linters are passing
- You are following GitHub flow
- Good commit history

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 4 above)
Comment (ekenecf) [reply app.rb:66]: Hi @keneogu, according to the project requirement, we were asked to create either student or teacher and I have implemented the student method as requested.

File: app.rb (hunk 3 above)
Comment (ekenecf) [reply app.rb]: Hi @keneogu, I have made the changes you asked me to make.

File: app.rb (hunk 4 above)
Comment (thecodechaser) [inline app.rb:66]: - [ ] As requested by the previous reviewer, Please add the `create_teacher` method. You need to create both methods to create students and teachers. In the project requirements, it's mentioned that there should be a `create_person` method that will be responsible to create students or teachers based on the user's choice.

### See the image for reuiqments

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @ekenecf 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    gets.chomp
+  end
+
+  def menu_run
+    choice = menu.to_s
+    case choice
+    when '1'
+      list_all_books
+    when '2'
+      list_all_person
```
Comment (FrederickMih) [inline app.rb]: - [ ] Hi @ekenecf, great job for implementing the changes suggested by the previous reviewers :+1: . I think your app breaks after listing a person or a book. You will once more make sure that your app behaves like the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) provided to you.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @ekenecf , 



Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
# Status: Approved :tada: :tada: :champagne: 

Hi @ekenecf , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 





Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**